docs/extending/rich_text_internals.md

```diff
@@ -197,6 +197,31 @@ def register_embed_handler(features):
     features.register_embed_type(MyCustomEmbedHandler)
 ```
 
+### Registering generic rewrite handlers
+
+If you need to add a new rich text feature that does not fit the pattern of the LinkHandler or the EmbedHandler, you can register an entirely new rewriter alongside your converter rule.
+
+Rewriters must be callables, that receive a single argument - the rich-text HTML - and are registered with the `register_frontend_rewriter` call on the rich-text `features` object. By convention they are written as classes with a `__call__` method, but a function would also suffice for simple replacers.
+
+```python
+class GreenRewriter:
```

Comment:
Did you consider making an example based on the use case you describe of tagging content in different languages? I think green is ok but the more grounded in real-world needs the better (#4694).

Reply:
Done, in concert with above. Example is now a `<maori>` tag, as this is our use-case